servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcServerBuilder.java

```diff
@@ -55,8 +56,15 @@
  */
 public abstract class GrpcServerBuilder {
 
+    @FunctionalInterface
+    public interface HttpInitializer {
+        void initialize(HttpServerBuilder builder);
+    }
+
     private boolean appendedCatchAllFilter;
 
+    public abstract GrpcServerBuilder initializer(HttpInitializer initializer);
```

Comment:
Does this affect the order of applied configurations? In other words, when we are in this transient state with the deprecations, if users call `appendHttpServiceFilter` directly (deprecated methods) and they also start using the `initializer` where they also might append filters, is the order of the filters consistent to what it would have been if that new API wasn't there? If thats the case, can we prevent bad configuration, e.g. allowing exclusively one way or the other.

Reply:
Great observation! To answer – the order would not be preserved as the `initializer` is applied at build time, while the deprecated methods have an immediate effect on the underlying builder. We could add documentation, but that would not prevent users from possible issues. I will try to implement a mutually exclusive mechanism, bear with me.

Reply:
Ok, I found where this comes from :) 
We should not be extra caution in this scenario. Users either migrate all `appendHttpServiceFilter` methods or nothing. If they use a mixed approach - this is their problem. We can clarify it in javadoc, but no need to add extra guards in the code. Neither multi-address nor partitioned client builders have similar guards.